### PR TITLE
HYPRE_USING_GPU

### PR DESCRIPTION
**Pull request by liruipeng**
This PR (by @pbauman #292 ) takes several instances of 
```
#if defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_DEVICE_OPENMP)
``` 
and changes them to 
```
#if defined(HYPRE_USING_GPU). 
```
This is not totally exhaustive: I only chose ones that were (seemingly...) very obvious. This is a precursor to adding HIP support. I tested this PR on V100 with CUDA 10.2 by compiling HYPRE with CUDA support and building the ij test and running it.

### CODE REVIEW

**Comment by liruipeng** [comment]
Hi @rfalgout, do you know why I can't add @pbauman as reviewer? Thanks!

**Comment by liruipeng** [comment]
Hi, @pbauman I merged the PR to this branch. Please push further changes to this one. There are a few places where we still have `#if defined(HYPRE_USING_CUDA) || defined(HYPRE_USING_DEVICE_OPENMP)`. I assume you don't want to change them since HIP is not ready for them. That's fine for us now.

To All: Now when we put code inside `#if defined(HYPRE_USING_GPU)`, we need to realize it needs to work for CUDA, OpenMP target, and HIP (and maybe more other architectures we are going to support).

All the changes look good to me! I will merge to master when you tell me (after running regressions tests). I won't make changes since I don't know how the changes will affect your code.

Thanks for cleaning these up.

-Ruipeng

**Comment by pbauman** [comment]
> I assume you don't want to change them since HIP is not ready for them. That's fine for us now.

Yes, please, if y'all don't mind. I'm most of the way through the first basic HIP pass. Anything after that regarding `HYPRE_USING_GPU` will be clean-up or purposefully left out because of compiler problems.

>  will merge to master when you tell me (after running regressions tests). 

Please merge when you're ready - my current basic HIP branch is built on top of this one, so it's good as far as I'm concerned.

**Comment by rfalgout** [comment]
Hi @liruipeng .  I just added @pbauman .  When you search for a person to add, I think you need to leave the '@' off of the name.  Maybe that's why you had trouble?

**Review by rfalgout — APPROVED** [review]
Looks good to me.  Thanks!

**Comment by rfalgout** [comment]
Hi @pbauman .  You should have write access to branches as a contributor, so if you need to make changes to this PR, you can do that on the HYPRE_USING_GPU branch and push them.  As Ruipeng pointed out in his email, our workflow is summarized here:

   https://github.com/hypre-space/hypre/wiki/Git-Workflow

Thanks again!

**Review by ulrikeyang — APPROVED** [review]
Looks fine to me.

**Review by pbauman — APPROVED** [review]

**Pull request merged**
